sqlmesh/core/analytics/__init__.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import atexit
+import os
+import typing as t
+from functools import wraps
+
+from sqlmesh.core.analytics.collector import AnalyticsCollector
+from sqlmesh.core.analytics.dispatcher import AsyncEventDispatcher, NoopEventDispatcher
+from sqlmesh.utils import str_to_bool
+
+
+def init_collector() -> AnalyticsCollector:
+    dispatcher = (
+        NoopEventDispatcher()
+        if str_to_bool(os.getenv("SQLMESH__DISABLE_ANONYMIZED_ANALYTICS", "false"))
+        else AsyncEventDispatcher()
+    )
+    return AnalyticsCollector(dispatcher=dispatcher)
+
+
+collector = init_collector()
+
+
+atexit.register(collector.shutdown, flush=True)
+
+
+def disable_analytics() -> None:
```

Comment:
why did you choose to do this pattern? should collector instead be a constant like COLLECTOR, which can then be disabled?

Reply:
This allowed me to reuse existing constructs (eg dispatcher) without explicitly adding `disable` API to the collector. I don't feel strongly about either approach. Do you see any advantages to doing this the way you suggested?

Reply:
no